src/bloqade/noise/native/stmts.py

```diff
@@ -7,25 +9,43 @@
 from ._dialect import dialect
 
 
-@statement(dialect=dialect)
-class PauliChannel(ir.Statement):
-
+@statement
+class NativeNoiseStmt(ir.Statement):
     traits = frozenset({lowering.FromPythonCall()})
 
+    @property
+    def probabilities(self) -> Tuple[Tuple[float, ...], ...]: ...
```

Comment:
if this is a required statement to implement I would raise a NotImplementedError here